exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java

```diff
@@ -104,26 +109,42 @@ private JaegerGrpcSpanExporter(String serviceName, ManagedChannel channel, long
    */
   @Override
   public CompletableResultCode export(Collection<SpanData> spans) {
-    Collector.PostSpansRequest request =
-        Collector.PostSpansRequest.newBuilder()
-            .setBatch(
-                Model.Batch.newBuilder()
-                    .addAllSpans(Adapter.toJaeger(spans))
-                    .setProcess(this.process)
-                    .build())
-            .build();
-
     CollectorServiceGrpc.CollectorServiceFutureStub stub = this.stub;
     if (deadlineMs > 0) {
       stub = stub.withDeadlineAfter(deadlineMs, TimeUnit.MILLISECONDS);
     }
 
+    Map<Resource, List<SpanData>> resourceAndSpanData = Adapter.groupByResource(spans);
+    List<Collector.PostSpansRequest> requests = new ArrayList<>();
+    for (Map.Entry<Resource, List<SpanData>> entry : resourceAndSpanData.entrySet()) {
+      Process.Builder builder = this.processBuilder.clone();
+      if (entry.getKey().getAttributes() != null) {
+        builder.addAllTags(Adapter.toKeyValues(entry.getKey().getAttributes()));
+      }
+
+      Collector.PostSpansRequest request =
+          Collector.PostSpansRequest.newBuilder()
+              .setBatch(
+                  Model.Batch.newBuilder()
+                      .addAllSpans(Adapter.toJaeger(entry.getValue()))
+                      .setProcess(builder.build())
+                      .build())
+              .build();
```

Comment:
I'd extract lines 120-132 into a separate method, and then you could use a java 8 Stream/map to collapse this logic significantly.

Reply:
Also, please add a unit test to the `JaegerGrpcSpanExporterTest` to cover this change.

Reply:
done

Reply:
I think it would be good to have a test that uses multiple resources, so we can verify that multiple batches are sent, and we don't just use the first Resource we find. I didn't see this case in the test code. did I miss it?

Reply:
added test